### PR TITLE
added PID class

### PR DESCRIPTION
**Pull request by BBScholar**

### CODE REVIEW

File: src/main/kotlin/org/team5499/monkeyLib/math/pid/PID.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        this.kD = kD
+        this.inverted = inverted
+    }
+
+    public constructor(): this(0.0, 0.0, 0.0, false)
+
+    public fun calculate(): Double {
+        accumulator += error * timer.get()
+        val pt = kP * error
+        val it = kI * accumulator
```
Comment (andycate) [inline src/main/kotlin/org/team5499/monkeyLib/math/pid/PID.kt]: You might consider adding an option for integral window, where the accumulator only remembers the values from the past `n` loop iterations

File: src/main/kotlin/org/team5499/monkeyLib/math/pid/PID.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    public fun reset() {
+        timer.stop()
+        timer.reset()
+        accumulator = 0.0
+        lastError = 0.0
+        setpoint = 0.0
+    }
+}
```
Comment (andycate) [inline src/main/kotlin/org/team5499/monkeyLib/math/pid/PID.kt]: wait, did you delete PIDF? Why?

Comment (BBScholar) [reply src/main/kotlin/org/team5499/monkeyLib/math/pid/PID.kt]: I dont know how to add the F term? do you just add a constant

File: src/main/kotlin/org/team5499/monkeyLib/math/pid/PID.kt (hunk 1 above)
Comment (BBScholar) [reply src/main/kotlin/org/team5499/monkeyLib/math/pid/PID.kt]: ok

File: src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt [hunk 3]
```diff
@@ -1,2 +1,94 @@
...
+    public var kP: Double
+    public var kI: Double
+    public var kD: Double
+    public var kF: Double
+    public var setpoint: Double
+    public var processVariable: Double
+
+    // number of iterations the accumulator keeps track of.
+    // 0 disables this (keeps track of all loops)
+    public var integralZone: Int
```
Comment (connorworley) [inline src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt]: What's the use case for this? The I term can't really do it's job unless it's allowed to grow unbounded.

File: src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt [hunk 4]
```diff
@@ -1,2 +1,94 @@
...
+        val ft = kF * setpoint
+        timer.stop()
+        timer.reset()
+        timer.start()
+        lastError = error
+        val total = pt + it + dt + ft
+        if (useOutputCaps) {
+            return if (!inverted) Utils.limit(total, lowerOutputCap, upperOutputCap)
+            else -Utils.limit(total, lowerOutputCap, upperOutputCap)
+        } else return total
```
Comment (connorworley) [inline src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt]: Personally I think it's better to make "lighter" APIs - if the user wants to add output limits or inversion they can do it themselves - they're already using MonekyLib anyways. Less stuff to write tests for too.

Comment (BBScholar) [reply src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt]: I agree.

File: src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt (hunk 3 above)
Comment (BBScholar) [reply src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt]: andy wanted it.

Comment (connorworley) [reply src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt]: Ok, feel free to ignore.

File: src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt (hunk 4 above)
Comment (BBScholar) [reply src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt]: Done

File: src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt (hunk 3 above)
Comment (BBScholar) [reply src/main/kotlin/org/team5499/monkeyLib/math/pid/PIDF.kt]: ok

**Pull request merged**